packages/core/utils/src/modules-sdk/define-link.ts

```diff
@@ -0,0 +1,314 @@
+import { LinkModulesExtraFields, ModuleJoinerConfig } from "@medusajs/types"
+import { isObject, pluralize, toPascalCase } from "../common"
+import { composeLinkName } from "../link"
+
+type InputSource = {
+  serviceName: string
+  field: string
+  linkable: string
+  primaryKey: string
+}
+
+type InputToJson = {
+  toJSON(): InputSource
+}
+
+type InputOptions = {
+  source: InputToJson | InputSource
```

Comment:
very nit: Think we should try to come up with a name closer to links, modules, and models. Perhaps, we could do `link` or `model`?

```ts
defineLink({ link: ProductModule.links.product, ...
defineLink({ model: ProductModule.links.product, ...
```